packages/medusa/src/api/routes/admin/draft-orders/create-line-item.ts

```diff
@@ -108,15 +121,43 @@ export default async (req, res) => {
     }
 
     if (validated.variant_id) {
-      const line = await lineItemService
-        .withTransaction(manager)
-        .generate(
-          validated.variant_id,
-          draftOrder.cart.region_id,
-          validated.quantity,
-          { metadata: validated.metadata, unit_price: validated.unit_price }
+      let line
+
+      if (
+        featureFlagRouter.isFeatureEnabled(IsolateProductDomainFeatureFlag.key)
+      ) {
+        const remoteQuery = req.scope.resolve("remoteQuery")
+        const [variant] = await retrieveVariantsWithIsolatedProductModule(
+          remoteQuery,
+          [validated.variant_id]
         )
 
+        if (!variant) {
+          throw new MedusaError(
+            MedusaError.Types.INVALID_DATA,
+            `Variant with id: ${validated.variant_id} not found`
+          )
+        }
```

Comment:
todo: I believe it is handled by the utils below, should we remove that?

Reply:
yep handled in the util as well...removed